src/ElmTestRunner/Result.elm

```diff
@@ -13,80 +13,128 @@ module ElmTestRunner.Result exposing
 
 # Helper functions
 
-@docs Summary, summary
+@docs Summary, summary, encodeCoverageReport
 
 -}
 
 import Array exposing (Array)
+import Dict exposing (Dict)
 import ElmTestRunner.Failure as Failure exposing (Failure)
 import Expect exposing (Expectation)
 import Json.Decode as Decode exposing (Decoder, Value)
 import Json.Encode as Encode
+import Test.Coverage exposing (CoverageReport(..))
 import Test.Runner
 
 
 {-| Type summarizing the results of a test run.
 It is obtained from the list of expectations returned by calling runner.run ().
 -}
 type TestResult
-    = Passed { labels : List String, duration : Float, logs : List String }
-    | Failed { labels : List String, duration : Float, logs : List String, todos : List String, failures : List Failure }
+    = Passed
+        { labels : List String
+        , duration : Float
+        , logs : List String
+        , successes : List CoverageReport -- TODO change to success : CoverageReport
+        }
+    | Failed
+        { labels : List String
+        , duration : Float
+        , logs : List String
+        , todos : List String
+        , failures : List ( Failure, CoverageReport )
+        }
 
 
 {-| Set the duration that the test took.
 -}
 setDuration : Float -> TestResult -> TestResult
 setDuration duration testResult =
     case testResult of
-        Passed { labels, logs } ->
```

Comment:
These records were fully expanded because I learned that this makes a lot performance difference. Maybe not worth it?

Reply:
I wasn't aware it influences performance. I'll revert back to your style

Reply:
Reverted in ed2dac9